### PR TITLE
Rename lengthscale to lengthscales

### PR DESCRIPTION
**Pull request by vdutor**

### CODE REVIEW

File: doc/source/notebooks/advanced/kernels.pct.py [hunk 1]
```diff
@@ -90,10 +90,10 @@ def plotkernelsample(k, ax, xmin=-3, xmax=3):
 # %% [markdown]
 # ## First example: create a Matern 3/2 covariance kernel
 #
-# Many kernels have hyperparameters, for example `variance` and `lengthscale`. You can change the value of these parameters from their default value of `1.0`.
+# Many kernels have hyperparameters, for example `variance` and `lengthscales`. You can change the value of these parameters from their default value of `1.0`.
```
Comment (st--) [inline doc/source/notebooks/advanced/kernels.pct.py]: Would it be helpful to add a line here explaining how lengthscales as a float gives you *isotropic* lengthscale (for higher-dimensional input) and to pass in an *array of floats* to get ARD?

Comment (st--) [reply doc/source/notebooks/advanced/kernels.pct.py]: Actually this is covered further down in the notebook

File: doc/source/notebooks/advanced/kernels.pct.py [hunk 2]
```diff
@@ -159,13 +159,13 @@ def plotkernelfunction(k, ax, xmin=-3, xmax=3, other=0):
 # %% [markdown]
 # ## Kernels for higher-dimensional input spaces
 #
-# Kernels generalize to multiple dimensions straightforwardly. Stationary kernels support "Automatic Relevance Determination" (ARD), that is, having a different lengthscale parameter for each input dimension. Simply pass in an array of the same length as the number of input dimensions. **NOTE:** This means that the kernel object is then able to process only inputs of that dimension!
+# Kernels generalize to multiple dimensions straightforwardly. Stationary kernels support "Automatic Relevance Determination" (ARD), that is, having a different lengthscales parameter for each input dimension. Simply pass in an array of the same length as the number of input dimensions. **NOTE:** This means that the kernel object is then able to process only inputs of that dimension!
```
Comment (st--) [inline doc/source/notebooks/advanced/kernels.pct.py]:
```suggestion
# Kernels generalize to multiple dimensions straightforwardly. Stationary kernels support "Automatic Relevance Determination" (ARD), that is, having a different lengthscale parameter for each input dimension. Simply pass in an array of the same length as the number of input dimensions. **NOTE:** This means that the kernel object is then able to process only inputs of that dimension!
```
Erroneous search&replace in this case I think

File: doc/source/notebooks/advanced/mcmc.pct.py [hunk 3]
```diff
@@ -217,13 +217,13 @@ def plot_joint_marginals(samples, y_axis_label):
     axs[0].set_ylabel('signal_variance')
 
     axs[1].plot(samples[name_to_index['.likelihood.variance']],
-                samples[name_to_index['.kernel.lengthscale']], 'k.', alpha = 0.15)
+                samples[name_to_index['.kernel.lengthscales']], 'k.', alpha = 0.15)
     axs[1].set_xlabel('noise_variance')
-    axs[1].set_ylabel('lengthscale')
+    axs[1].set_ylabel('lengthscales')
```
Comment (st--) [inline doc/source/notebooks/advanced/mcmc.pct.py]:
```suggestion
    axs[1].set_ylabel('lengthscale')
```

File: doc/source/notebooks/advanced/mcmc.pct.py [hunk 4]
```diff
@@ -217,13 +217,13 @@ def plot_joint_marginals(samples, y_axis_label):
...
 
-    axs[2].plot(samples[name_to_index['.kernel.lengthscale']],
+    axs[2].plot(samples[name_to_index['.kernel.lengthscales']],
                 samples[name_to_index['.kernel.variance']], 'k.', alpha = 0.1)
-    axs[2].set_xlabel('lengthscale')
+    axs[2].set_xlabel('lengthscales')
```
Comment (st--) [inline doc/source/notebooks/advanced/mcmc.pct.py]:
```suggestion
    axs[2].set_xlabel('lengthscale')
```

File: doc/source/notebooks/advanced/multiclass_classification.pct.py [hunk 5]
```diff
@@ -72,8 +72,8 @@
 C = 3
 N = 100
 
-# RBF kernel lengthscale
-lengthscale = 0.1
+# RBF kernel lengthscales
```
Comment (st--) [inline doc/source/notebooks/advanced/multiclass_classification.pct.py]:
```suggestion
# Lengthscale of the SquaredExponential kernel (isotropic -- change to `[0.1] * C` for ARD)
```
is this one supposed to be isotropic?

File: doc/source/notebooks/basics/GPLVM.pct.py [hunk 6]
```diff
@@ -83,11 +83,11 @@
 # %% [markdown]
 # We construct a Squared Exponential (SE) kernel operating on the two-dimensional latent space. 
 # The `ARD` parameter stands for Automatic Relevance Determination, which in practice means that
-# we learn a different lengthscale for each of the input dimensions. See [Manipulating kernels](../advanced/kernels.ipynb) for more information.
+# we learn a different lengthscales for each of the input dimensions. See [Manipulating kernels](../advanced/kernels.ipynb) for more information.
```
Comment (st--) [inline doc/source/notebooks/basics/GPLVM.pct.py]:
```suggestion
# we learn a different lengthscale for each of the input dimensions. See [Manipulating kernels](../advanced/kernels.ipynb) for more information.
```

File: doc/source/notebooks/basics/regression.pct.py [hunk 7]
```diff
@@ -179,7 +179,7 @@ def objective_closure():
 # %% [markdown]
 # ## GP regression in higher dimensions
 #
-# Very little changes when the input space has more than one dimension. By default, the `lengthscale` is an isotropic (scalar) parameter. It is generally recommended that you allow to tune a different lengthscale for each dimension (Automatic Relevance Determination, ARD): simply initialize `lengthscale` with an array of length $D$ corresponding to the input dimension of `X`.
+# Very little changes when the input space has more than one dimension. By default, the `lengthscales` is an isotropic (scalar) parameter. It is generally recommended that you allow to tune a different lengthscale for each dimension (Automatic Relevance Determination, ARD): simply initialize `lengthscale` with an array of length $D$ corresponding to the input dimension of `X`.
```
Comment (st--) [inline doc/source/notebooks/basics/regression.pct.py]:
```suggestion
# Very little changes when the input space has more than one dimension. By default, the `lengthscales` is an isotropic (scalar) parameter. It is generally recommended that you allow to tune a different lengthscale for each dimension (Automatic Relevance Determination, ARD): simply initialize `lengthscales` with an array of length $D$ corresponding to the input dimension of `X`.
```

Comment (st--) [reply doc/source/notebooks/basics/regression.pct.py]: should we also point to the advanced/kernels notebook here ?

File: doc/source/notebooks/theory/FITCvsVFE.py [hunk 8]
```diff
@@ -64,7 +64,7 @@ def getSparseModel(X, Y, isFITC=False):
 def printModelParameters(model):
     print("  Likelihood variance = {:.5g}".format(model.likelihood.variance.numpy()))
     print("  Kernel variance     = {:.5g}".format(model.kernel.variance.numpy()))
-    print("  Kernel lengthscale  = {:.5g}".format(model.kernel.lengthscale.numpy()))
+    print("  Kernel lengthscales  = {:.5g}".format(model.kernel.lengthscales.numpy()))
```
Comment (st--) [inline doc/source/notebooks/theory/FITCvsVFE.py]:
```suggestion
    print("  Kernel lengthscale  = {:.5g}".format(model.kernel.lengthscales.numpy()))
```

File: doc/source/notebooks/theory/Sanity_check.pct.py [hunk 9]
```diff
@@ -122,7 +122,7 @@ def plot(m, color, ax):
 # %%
 for m in models:
     print(m.__class__.__name__)
-    print(f"  kernel lengthscale  = {m.kernel.lengthscale.numpy():.5g}")
+    print(f"  kernel lengthscales  = {m.kernel.lengthscales.numpy():.5g}")
```
Comment (st--) [inline doc/source/notebooks/theory/Sanity_check.pct.py]:
```suggestion
    print(f"  kernel lengthscale  = {m.kernel.lengthscales.numpy():.5g}")
```

File: tests/gpflow/kernels/test_kernels.py [hunk 10]
```diff
@@ -375,25 +375,25 @@ def test_active_product(N, D):
 def test_ard_init_scalar(D):
     """
     For ard kernels, make sure that kernels can be instantiated with a single
-    lengthscale or a suitable array of lengthscale
+    lengthscales or a suitable array of lengthscales
```
Comment (st--) [inline tests/gpflow/kernels/test_kernels.py]:
```suggestion
    lengthscale or a suitable array of lengthscales
```

**Comment by vdutor** [comment]
Addressed your comments - thanks. I'll merge once the tests pass.

**Pull request merged**